Added dead marker.

NamStalker opened the pull request:
Added the dead marker rendering to the mod, using some math to scale up/down from distance. Limited to 10 dead markers on screen to not clutter it up too much. Created the dead marker icon in gimp using a pictures from issue as reference.

NamStalker committed:
Adding initial changes for the dead marker update.

NamStalker committed:
Adding initial changes for the dead marker update.

NamStalker committed:
Adding placeholder for dead marker file.

NamStalker committed:
Adding placeholder for dead marker file.

NamStalker committed:
Added newest dead marker, along with dead marker rendering. neads tweaking on size scaling from distance.

NamStalker committed:
Merge branch 'master' of https://github.com/NamStalker/GGOV

NamStalker committed:
Added final tweaks.

NamStalker committed:
Took out unneaded method, and using statement.

NamStalker commented:
#7

justalemon commented on GGO.Singleplayer/Main.cs:
```diff
@@ -75,6 +79,16 @@ private void OnTick(object Sender, EventArgs Args)
                     }
+
+                    // Check for on screen dead Peds to display dead markers, limit to 10 to not clutter the screen.
+                    if (NearbyPed.IsDead && NearbyPed.IsOnScreen && DeadCount < 10)
```
```suggestion
                    if (NearbyPed.IsDead && NearbyPed.IsOnScreen)
```
See the picture on #7, we want to show all of them.

justalemon commented on GGO.Singleplayer/Main.cs:
```diff
@@ -75,6 +79,16 @@ private void OnTick(object Sender, EventArgs Args)
+                        Draw.DeadMarker(Config, NearbyPed, DeadCount);
+
+                        // Increase the count of dead peds "rendered"
+                        DeadCount++;
```
```suggestion
```

SleepyNatDev replied:
Limit has been removed, though the counter is stilll needed to draw unique files to the UI screen, as it won't draw the same file multiple times.

justalemon replied:
Use the `Ped.Hash` property to make them unique. 
The hashes are unique per ped and we only need one image so that is ok.

SleepyNatDev replied:
I can add this, but I fear it may end up cluttering the %TEMP%/GGO folder. There will be a .png file for every ped hash. With the counter, it reuses the numbered files when drawing.

justalemon commented on GGO.Singleplayer/Main.cs:
```diff
@@ -57,6 +57,10 @@ private void OnTick(object Sender, EventArgs Args)
 
+                // Draw the dead markers
+                // Create a counter to limit the screen clutter.
+                int DeadCount = 0;
```
```suggestion
```

justalemon commented on GGO.Common/Properties/Resources.resx:
```diff
@@ -205,4 +205,10 @@
+  <data name="DeadMarker" type="System.Resources.ResXFileRef, System.Windows.Forms">
+    <value>..\Resources\DeadMarker.png;System.Drawing.Bitmap, System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a</value>
+  </data>
+  <data name="DeadMarker" type="System.Resources.ResXFileRef, System.Windows.Forms">
```
Twice?

SleepyNatDev replied:
That was weird, I didn't notice that. I fixed this as well though.

justalemon reviewed (changes requested):
Is good, but we need some changes

NamStalker committed:
Remove counter limit for marker

Removed counter for dead marker to show on all dead peds within the search radius. Counter still required for unique render of dead marker, one file will not draw multiple times.

NamStalker committed:
changed comment

NamStalker committed:
Merge pull request #1 from NamStalker/NamStalker-patch-1

Remove counter limit for marker

NamStalker committed:
Fixed duplicate dead marker in resx.

NamStalker committed:
Merge branch 'master' of https://github.com/NamStalker/GGOV

justalemon committed:
Merge branch 'master' into dead-markers

justalemon commented:
Looking good so far (I have a reduced, almost minimal, draw distance):
![9450629730161131520_20181018175828_1](https://user-images.githubusercontent.com/11861253/47183897-926e5d80-d2ff-11e8-8428-4cdcdc8a201d.png)

The only thing missing is to use the ped hashes and we are ready to go.

If you want, I can do what is missing.

justalemon committed:
Use the hashes for the image names
